### DISCUSSION TITLE
Улучшение разделов документации

### DISCUSSION THREAD

**Discussion by VadimOnix** [bol-van/zapret#359] [Ideas]
@bol-van приветствую! Я хочу помочь улучшить документацию в репозитории, у меня появилось свободное время на ближайшие пару месяцев. На своей работе я пишу много документации для разработчиков начального уровня.

На прошлой неделе впервые попробовал разобраться в том, как настроить данную утилиту на своём роутере и потратил значительно больше времени, чем могло бы потребоваться новичку. Я думаю, многие люди, которые не имеют достаточного опыта работы с linux системами, хотели бы настроить обход блокировок на своём оборудовании, но сталкиваются с проблемой в виде большого объёма текста Readme и не могут эффективно воспользоваться поиском по документу, потому что даже не знают, что именно им нужно.

Как ты смотришь на то, что я опишу структуру документации в данном обсуждении, ты её посмотришь, дашь комментарии, после чего в отдельной ветке через пулреквесты мы переработаем всю структуру документации?

Я видел https://github.com/bol-van/zapret/pull/217 но мне кажется этого недостаточно.

**Comment by bol-van** [comment]
я вижу проект как инструмент противодействия dpi больше для квалифицированной публики
у меня нет желания делать это простым и дрступныи
и не хочу превращения доки из образовательно технического
описания в набор пошаговок для разной степени чайниковости
в рамках это концепции можете высказать свои предложения

**Comment by bol-van** [comment]
считаю, что методика обмана дпи в принципе требует аналитического подхода
никакие пошаговки не дадут гарантированный результат
и создадут ложное впечатление, что можно выполнить 1 2 3...
и все должно заработать
а потом удивления то не открывается, здесь подвисает и тд
обслуживать такой контингент именно по вопросам я сделал 1 2 3 и не заработало
просто потому, что ситуация несколько сложнее, чем 1 2 3, без квалифицированного содействия у меня нет желания
это идет на "помощь зала"

**Reply by VadimOnix** [reply]
Я солидарен с позицией, что zapret - это не волшебный инструмент, который работает по принципу _"установил и забыл"_. Но в то же время, проанализировав дискуссии и пулреквесты, которые создаются в репозитории, я заметил, что большинству действительно достаточно вставить чужую готовую конфигурацию, чтобы получить доступ к условному YouTube или Rutracker. (_Такие посты набирают много лайков и порождают за собой ещё большее количество *необразованных* пользователей, которые могут навредить качеству своего интернет соединения_).

Моё предложение нацелено на то, чтобы превентивно снизить кол-во таких обсуждений, а также упростить жизнь тем, кто действительно хочет глубоко разобраться в возможностях утилиты. Документацию же в любом случае приятнее читать, когда она имеет хорошую навигацию и репрезентативное форматирование.

**Reply by bol-van** [reply]
про форматирование не спорю. мне однажды помогли сделать англ вариант в markdown
теперь так и веду

а по поводу содержания в чем суть вашей идеи ?

**Reply by VadimOnix** [reply]
1. Перейти на использование markdown
2. Изменить последовательность разделов рутового Readme.md и значительно сократить его объём
```mdx
-Введение
`Кратко описать, что это за инструмент, сейчас этот раздел уже готов`
- Установка
`Этот раздел содержит список ссылок на внешние md файлы в директории docs под каждую платформу`
-- Рекомендации по предварительной настройке
`Здесь перечисляется тезисно, что было бы неплохо настроить на устройстве, чтобы повысить шансы работы запрета. Например, в моём случае - это была установка пакета dnscrypto-proxy на Ubuntu, можно оставить сслыки на их официальную документаицю по настройке, возможно для Windows есть аналогичный набор утилит.`
- Технологии обхода
`Здесь тезисно описано, что инструмент имеет разные методы обхода блокировок, в корневом разделе, кажется, основные это tpws и nfqws, про них есть смысл рассказать сразу`
-- TPWS
-- NFQWS
`В каждом из этих разделов описывается суть метода (теоретическое обоснование). и ссылки на внешние документы, в которых подробно описаны параметры, которые можно использовать в файле конфигурации`
- FAQ
`Думаю, что на основе множества дискуссий в репе уже можно сформировать список вопросов по типу "Можно ли устанавливать утилиту с параметрами по-умолчанию?" - нет, запустите blockchek.sh и дождитесь пока скрипт подберёт для вас оптимальную стратегию;  "Можно ли установить на мой роутер? - ссылка на таблицу совместимости маршрутизаторов OpenWRT;  "Сколько постоянной памяти рекомендуется для режима hostlist?" и т.д. Я сейчас выдумываю из головы, но можно будет провести анализ`
```
3. Разделить корневой файл на множество поменьше и группировать их в зависимости от контекста.

Я понимаю, что корневой документ кропотливо заполнялся месяцами, и вы как автор знаете его вдоль и поперёк, но человек, который первый раз приходит в репозиторий, должен иметь возможность погружаться в технологию настолько - насколько требует поставленная перед ним задача. 
Удалять, я думаю, ничего не потребуется, но разделить всю информацию по разделам будет отличной идеей!

**Reply by bol-van** [reply]
такая структура может была бы полезна для продукта, где 
одни и те же действия приводят к одному и тому же результату
но здесь это же не так
можно расписать таким способом как установить систему
но не как заставить ее выдавать желаемое
а тут уже без знаний сложно
в принципе я не против, но этот момент должен проходить красной линией
чтобы чел сразу понимал, что это лишь помощь в установке
но ключики подбирать ему придется самому

еще я против советов "на всякий случай"
типа установить днс крипт
каждое действие должно иметь показания к применению
если днс перехвачен провом тогда да
а иначе нет
и юзер должен с этим разобраться надо или нет в его случае

блокчек тоже не панацея. он лишь дает чем  можно обойти конкретный сайт
но если тупо копировать скорее всего результат будет не очень
многое сломается или не обойдется

и будет еще больше вопросов. только теперь уже к автору
тк типа он составил этот faq

**Comment by numbereleven-a** [comment]


**Reply by bol-van** [reply]
до определенного предела
если я вижу, что чел совсем не сечет тему
или не может внятно обьяснить техническую сторону
я отправляю на "помощь зала"

**Reply by numbereleven-a** [reply]
ну а будь документация более понятная для людей которые линукс видят первый раз, может меньше не поняток будет, будешь меньше отвлекаться)

**Reply by bol-van** [reply]
вопрос с нубами решается просто. игнор или отсыл в дискуссии. чтобы нубы сами помогали себе. здесь не детский сад. захотят научатся. а нет - так им тут нечего делать.
разжевывать для идиотов - дело неблагодарное
всегда найдется еще больший идиот, кому и этого будет мало
как написали на 4пда, так можно дойти до инструкции как вставлять кабель в роутер
тем самы потеряется суть проекта

**Comment by BubaMuba** [comment]
Пожалуй, соглашусь с автором насчёт отношения к документации. Незачем метать бисер перед такими свиньями. как zifirik.

**Reply by numbereleven-a** [reply]
> Пожалуй, соглашусь с автором насчёт отношения к документации. Незачем метать бисер перед такими свиньями. как zifirik.

тут какой то отдельный случай всё таки

**Reply by bol-van** [reply]
да, тут явно что-то не то с головой. товарищу надо как мин опохмел.

но в целом я бы разделил категории юзеров

1.  технически грамотные. способные понять как это работает по большей части. это основной таргет
2.  не особо прошареные в сетях, но знают ос и сети на базовом уровне. нянька как скопировать не нужна. готовы до какой-то степени вникать.  тут помогут факи и инструкции, но не до разжевывания нажмите тут и там. без потери смысла действий
3. дяди васи, видящие линух 3-й раз в жизни. этим я не особо хочу помогать. пусть идут в сообщества их уровня. там может быть разжуют
4. неадекваты, считающие, что им тут обязаны были в рот положить. в места отдаленные таких. особо отличившихся в бан

**Comment by neduus** [comment]
Тут нужна не документация, а скорее packetflow diagram в дополнение к уже существующей документации.
По картинке будет проще понимать, какие параметры на что влияют. Аля quic шпаргалка.
